api/operator/v1beta1/vmalertmanager_types.go

```diff
@@ -484,8 +490,12 @@ func (cr *VMAlertmanager) ProbePort() string {
 	return cr.Spec.PortName
 }
 
+// ProbeScheme returns scheme for probe
 func (cr *VMAlertmanager) ProbeScheme() string {
-	return strings.ToUpper(protoFromFlags(cr.Spec.ExtraArgs))
+	if cr.Spec.WebConfig != nil && cr.Spec.WebConfig.TLSServerConfig != nil {
```

Comment:
not sure that this validation is enough. user can set `WebserverTLSConfig.ClientAuthType` to `NoClientCert`

Reply:
is there any validation on alertmanager site?

Reply:
In this case, tls must be configured. 

 This PR lacks of:
- crd server-side validation
- validation webhook checks
- mention of changes at changelog

It should be addressed before merging.

Reply:
updated checks, added validation and changelog entry